### ISSUE TITLE
crossfade() breaks RequestDisposable.await()

### ISSUE THREAD

**Issue by orwir** [coil-kt/coil#241]
**Describe the bug**
crossfade() breaks the RequestDisposable.await() method then scroll up in the RecyclerView. Image is loaded properly but await() has never return control back.
* Scroll down works as expected.
* Scroll up without `crossfade()` works as expected.

**To Reproduce**
1. create method like this to load images into views and hide progress when `true` is collected.
```kotlin
    fun ImageView.load(url: String): Flow<Boolean> = flow {
        emit(false)
        loader.load(context, url) {
            crossfade(true)
            target(this@load)
        }.await()
        emit(true)
    }
```
2. Scroll down enough and then start to scroll up.
3. flow has never emitted `true` because .await() wouldn't return control back

**Version**
0.9.1
Android 10 (29)

**Comment by colinrtwhite** [comment]
Thanks for the report. The issue is `CrossfadeTransition` is suspending indefinitely for a memory cache event. This was just fixed by https://github.com/coil-kt/coil/pull/240 and will be released as part of `0.9.2` very soon. In the meantime you can depend on [the latest snapshot](https://coil-kt.github.io/coil/faq/) to get the fix immediately.

**Comment by orwir** [comment]
Thanks a lot for your answer. I've just removed crossfade for now. Will wait a new release.
